tests/basis_test.py

```diff
@@ -3,328 +3,296 @@
 import numpy
 from scipy import special
 
-from smolyay.basis import (BasisFunction, ChebyshevFirstKind,
-        ChebyshevSecondKind, BasisFunctionSet, NestedBasisFunctionSet)
-
-@pytest.fixture
-def expected_points_3_set():
-    """extrema for exactness = 8"""
-    return [0, -1.0, 1.0, -1/numpy.sqrt(2), 1/numpy.sqrt(2),
-            -numpy.sqrt(numpy.sqrt(2)+1)/(2**0.75),
-            -numpy.sqrt(numpy.sqrt(2)-1)/(2**0.75),
-            numpy.sqrt(numpy.sqrt(2)-1)/(2**0.75),
-            numpy.sqrt(numpy.sqrt(2)+1)/(2**0.75)]
-
-def test_cheb_initial_zero():
-    """test degree of zero"""
-    f = ChebyshevFirstKind(0)
-    assert f.n == 0
-    assert f.points == [0]
-
-def test_cheb_initial_1():
-    """test initial when degree is 1"""
-    f = ChebyshevFirstKind(1)
-    assert f.n == 1
-    assert f.points == [-1, 1]
-
-def test_cheb_initial_2():
-    """test initial when degree is 2"""
-    f = ChebyshevFirstKind(2)
-    assert f.n == 2
-    assert numpy.allclose(f.points,[-1, 0, 1],atol=1e-10)
-
-def test_cheb_initial_4():
-    """test initial when degree is 4"""
-    expected_points = [-1, -1/numpy.sqrt(2), 0, 1/numpy.sqrt(2), 1]
-    f = ChebyshevFirstKind(4)
-    assert f.n == 4
-    assert numpy.allclose(f.points,expected_points,atol=1e-10)
-
-def test_cheb_initial_8():
-    """test initial when degree is 8"""
-    expected_points = [-1, -numpy.sqrt(numpy.sqrt(2)+1)/(2**0.75),
-            -1/numpy.sqrt(2), -numpy.sqrt(numpy.sqrt(2)-1)/(2**0.75), 0,
-            numpy.sqrt(numpy.sqrt(2)-1)/(2**0.75), 1/numpy.sqrt(2),
-            numpy.sqrt(numpy.sqrt(2)+1)/(2**0.75), 1]
-    f = ChebyshevFirstKind(8)
-    assert f.n == 8
-    assert numpy.allclose(f.points,expected_points,atol=1e-10)
-
-def test_cheb_call_degree_0_1():
-    """Chebyshev polynomial degree 0 is always 1 and degree 1 returns input"""
-    f0 = ChebyshevFirstKind(0)
-    f1 = ChebyshevFirstKind(1)
-    for i in [-1, -0.5, 0, 0.5, 1]:
-        assert f0(i) == 1
-        assert f1(i) == i
-
-def test_cheb_call_random_points():
-    """Test chebyshev polynomial at some degree at some input"""
-    numpy.random.seed(567)
-    ns = numpy.random.randint(20,size = 20)
-    xs = numpy.random.rand(20) * 2 - 1
-    for n,x in zip(ns,xs):
-        f = ChebyshevFirstKind(n)
-        assert numpy.isclose(f(x),special.eval_chebyt(n,x))
+import smolyay.basis
```

Comment:
Just import `smolyay`
```suggestion
import smolyay
```
If it isn't working like that, then we should fix the import structure of the package. Let me know

Reply:
It doesn't work.

Reply:
We must be missing imports in `__init__.py`. In that file, add:
```py
from . import basis
```
You should do this for all module files to bind them to the package namespace. See here for more information:

https://docs.python.org/3/reference/import.html#package-relative-imports